### PR TITLE
fix: arrayfield items prop fallback to empty array when record is null

### PR DESCRIPTION
**Pull request by Jshhhh**
*Issue #, if available:*

*Description of changes:*
- Fix ArrayField component when DS array type record value is null

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/codegen-ui-react/lib/__tests__/__snapshots__/studio-ui-codegen-react-forms.test.ts.snap [hunk 1]
```diff
@@ -2036,7 +2036,7 @@ export default function NestedJson(props) {
         }}
         currentFieldValue={currentNicknamesValue}
         label={\\"nick-Names2\\"}
-        items={nicknames}
```
Comment (hein-j) [inline packages/codegen-ui-react/lib/__tests__/__snapshots__/studio-ui-codegen-react-forms.test.ts.snap:2039]: 1. shouldn't the variable name still be `nicknames`?
2. what do you think about doing this in the reset function instead? Something like `setNicknames(cleanedValue.nickNames ?? [])`. If too big of a change, maybe not worth it.

File: packages/codegen-ui-react/lib/utils/forms/array-field-component.ts [hunk 2]
```diff
@@ -1062,7 +1065,14 @@ export const renderArrayFieldComponent = (
         ),
         factory.createJsxAttribute(
           factory.createIdentifier('items'),
-          factory.createJsxExpression(undefined, factory.createIdentifier(renderedFieldName)),
+          factory.createJsxExpression(
+            undefined,
+            factory.createBinaryExpression(
+              factory.createIdentifier('items'),
```
Comment (awinberg-aws) [inline packages/codegen-ui-react/lib/utils/forms/array-field-component.ts:1071]: should this be `renderedFieldName`?

**Review by awinberg-aws — CHANGES_REQUESTED** [review]

File: packages/codegen-ui-react/lib/utils/forms/array-field-component.ts (hunk 2 above)
Comment (Jshhhh) [reply packages/codegen-ui-react/lib/utils/forms/array-field-component.ts:1071]: Yeah it should thanks

File: packages/codegen-ui-react/lib/__tests__/__snapshots__/studio-ui-codegen-react-forms.test.ts.snap (hunk 1 above)
Comment (Jshhhh) [reply packages/codegen-ui-react/lib/__tests__/__snapshots__/studio-ui-codegen-react-forms.test.ts.snap:2039]: 1. Yes it should
2. Yeah thats is possible

**Review by hein-j — COMMENTED** [review]
We don't have to have the fallback if `initialValues`, not `cleanValues` because that means it's not getting from DataStore. And we already set the initial value of ArrayFields to an empty array.

**Comment by Jshhhh** [comment]
> We don't have to have the fallback if `initialValues`, not `cleanValues` because that means it's not getting from DataStore. And we already set the initial value of ArrayFields to an empty array.

The array type field returned from DataStore can be null so the array type field state would be set to `null`, which would break ArrayField component

**Review by hein-j — APPROVED** [review]

**Review by awinberg-aws — APPROVED** [review]

**Pull request merged**